tests/components/border_editor.test.ts

```diff
@@ -0,0 +1,298 @@
+import { App, Component, onMounted, onWillUnmount, useState, useSubEnv, xml } from "@odoo/owl";
+import { TopBar } from "../../src/components/top_bar/top_bar";
+import { DEFAULT_BORDER_DESC } from "../../src/constants";
+import { Model } from "../../src/model";
+import { OWL_TEMPLATES } from "../setup/jest.setup";
+import { selectCell, setSelection } from "../test_helpers/commands_helpers";
+import { simulateClick } from "../test_helpers/dom_helper";
+import { getBorder } from "../test_helpers/getters_helpers";
+import { makeTestFixture } from "../test_helpers/helpers";
+
+jest.mock("../../src/components/composer/content_editable_helper", () =>
+  require("./__mocks__/content_editable_helper")
+);
+jest.mock("../../src/helpers/uuid", () => require("../__mocks__/uuid"));
+
+let fixture: HTMLElement;
+const t = (s: string): string => s;
+
+async function setBorder(name: string) {
+  await simulateClick('.o-dropdown[title="Borders"]');
+  await simulateClick(`.o-line-item[name="${name}"]`);
+}
+
+class Parent extends Component {
+  static template = xml/* xml */ `
+    <div class="o-spreadsheet">
+      <TopBar focusComposer="state.focusComposer" onClick="() => {}"/>
+    </div>
+  `;
+  static components = { TopBar };
+
+  static _t = t;
+  state = useState({ focusComposer: <boolean>false });
+
+  setup() {
+    useSubEnv({
+      openSidePanel: () => {},
+      model: this.props.model,
+      askConfirmation: jest.fn(),
+      _t: Parent._t,
+      isDashboard: () => this.props.model.getters.isDashboard(),
+    });
+    this.state.focusComposer = this.props.focusComposer || false;
+    onMounted(() => this.props.model.on("update", this, this.render));
+    onWillUnmount(() => this.props.model.off("update", this));
+  }
+
+  setFocusComposer(isFocused: boolean) {
+    this.state.focusComposer = isFocused;
+  }
+}
+
+async function mountParent(
+  model: Model = new Model(),
```

Comment:
use the new helper  `mountComponent, it will reduce a lot of the prep work you are doing